File: Easy/459. Repeated Substring Pattern/459. Repeated Substring Pattern.js
/**
 * @param {string} s
 * @return {boolean}
 */
var repeatedSubstringPattern = function (s) {
    const n = s.length;

    for (let len = 1; len <= n / 2; len++) {
        if (n % len === 0) {
            const repeatCount = n / len;
            const pattern = s.substring(0, len);
            const constructed = pattern.repeat(repeatCount);
            if (constructed === s) {
                return true;
            }
        }
    }

    return false;
};

File: Easy/459. Repeated Substring Pattern/459. Repeated Substring Pattern.rs
impl Solution {
    pub fn repeated_substring_pattern(s: String) -> bool {
        let n = s.len();

        for len in 1..=n / 2 {
            if n % len == 0 {
                let repeat_count = n / len;
                let pattern = &s[0..len];
                let constructed = pattern.repeat(repeat_count);
                if constructed == s {
                    return true;
                }
            }
        }

        false
    }
}
